reference-implementation/lib/exclusive-stream-reader.js

```diff
@@ -11,17 +12,17 @@ export default class ExclusiveStreamReader {
       throw new TypeError('This stream has already been locked for exclusive reading by another reader');
     }
 
-    stream._readableStreamReader = this;
+    AttachExclusiveStreamReader(stream, this);
```

Comment:
Any reason not to inline this? It seems to be only used in one place. Same with detach. I don't feel strongly though so if you think it's clearer we can keep it.